modules/core/include/opencv2/core/utility.hpp

```diff
@@ -179,7 +179,7 @@ CV_EXPORTS ErrorCallback redirectError( ErrorCallback errCallback, void* userdat
 CV_EXPORTS String tempfile( const char* suffix = 0);
 CV_EXPORTS void glob(String pattern, std::vector<String>& result, bool recursive = false);
 
-/** @brief OpenCV will try to set the number of threads for the next parallel region.
+/** @brief OpenCV will try to set the number of threads for subsequent parallel regions.
 
 If threads == 0, OpenCV will disable threading optimizations and run all it's functions
```

Comment:
Proposed change looks good to me.

---

However there is another design issue. Which contradicts to the real implementation of some parallel backends.
There is mess with `threads == 0` and `threads == 1` values.

E.g., OpenMP case:

https://github.com/opencv/opencv/blame/913a2dbdaa3fbd011f7d4f08b381deb9dfe43b97/modules/core/src/parallel.cpp#L469

```
omp_set_num_threads(threads > 0 ? threads : numThreadsMax);
```

(code is from 2012 at least)

Docs states this:

> If threads == 0, OpenCV will disable threading optimizations and run all it's functions sequentially

So, using '0' parameter value is not recommended at all and it is a subject to the change.

---

Another important note, `setNumThreads()` is not a thread-safe function.
It is fatal to call it simultaneously in case of TBB at least (TBB's arena may become corrupted).